File: handlerclaw/lib/features/devices/data/mappers/user_device_mapper.dart
import 'package:handlerclaw/features/devices/domain/entities/user_device_entity.dart';
import 'package:handlerclaw/features/devices/data/dto/user_device_dto.dart';

class UserDeviceMapper {
  static UserDeviceEntity fromDto(UserDeviceDto dto) {
    return UserDeviceEntity(
      id: dto.id,
      userId: dto.userId,
      deviceId: dto.deviceId,
      deviceBrand: dto.deviceBrand,
      deviceModel: dto.deviceModel,
      osVersion: dto.osVersion,
      fcmToken: dto.fcmToken,
      platform: dto.platform,
      status: dto.status,
      lastSeenAt: dto.lastSeenAt,
      createdAt: dto.createdAt,
      updatedAt: dto.updatedAt,
    );
  }
}

File: api/src/modules/(jwt-auth)/user-devices/infrastructure/mappers/user-device.mapper.ts
import type { UserDevice } from "../../../../../lib/generated/prisma/client.js";
import type { IUserDevice } from "../../domain/entities/user-device.entity.js";

export function toUserDeviceEntity(data: UserDevice): IUserDevice {
  return {
    id: data.id,
    user_id: data.userId,
    status: data.status,
    device_id: data.deviceId,
    device_brand: data.deviceBrand,
    device_model: data.deviceModel,
    os_build_id: data.osBuildId,
    os_version: data.osVersion,
    fcm_token: data.fcmToken,
    platform: data.platform,
    last_seen_at: data.lastSeenAt,
    created_at: data.createdAt,
    updated_at: data.updatedAt,
  };
}
